src/conways.rs

```diff
@@ -0,0 +1,85 @@
+use std::{fmt::Display, usize};
+
+pub struct Conways {
+    world: Vec<Vec<u8>>,
+    width: usize,
+    height: usize
+}
+
+impl Conways {
+    pub fn new(width: usize, height: usize) -> Self {
+        Self {
+            world: vec![vec![0; width]; height],
+            width,
+            height
+        }
+    }
+
+    pub fn set_alive(&mut self, row: usize, column: usize) {
+        self.world[row][column] = 1;
+    }
+
+    pub fn tick(&mut self) {
+        let mut new_world: Vec<Vec<u8>> = vec![vec![0; self.width]; self.height];
+        for row in 0..self.height {
+            for column in 0..self.width {
+                let neighbours_count = self.neighbours_count(row, column);
+
+                let new_state = match (neighbours_count, self.world[row][column]) {
+                    // underpopulation
+                    (0..=1, 1) => 0,
+                    // survives
+                    (2..=3, 1) => 1,
+                    // overpopulation
+                    (4.., 1) => 0,
+                    // reproduction
+                    (3, 0) => 1,
+                    // no reproduction
+                    _ => 0 
+                };
```

Comment:
Have you considered using an enum instead of 1 and 0 to represent a cell's state?

IMO, I think it makes the code a bit easier to understand and gives you type safety!

PLUS, Rust's enums are super useful! What do you think?

Reply:
Yes, I think that would be better

Reply:
Done!